src/app/services/error-handler.service.ts

```diff
@@ -0,0 +1,249 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { ErrorHandler, Injectable, Injector } from '@angular/core';
+import { AbstractControl, UntypedFormGroup } from '@angular/forms';
+import { TranslateService } from '@ngx-translate/core';
+import * as Sentry from '@sentry/angular';
+import { Observable } from 'rxjs';
+import { ResponseErrorType } from 'app/enums/response-error-type.enum';
+import { sentryCustomExceptionExtraction } from 'app/helpers/error-parser.helper';
+import { ErrorReport } from 'app/interfaces/error-report.interface';
+import { Job } from 'app/interfaces/job.interface';
+import { WebsocketError } from 'app/interfaces/websocket-error.interface';
+import { DialogService } from 'app/services';
+
+@Injectable()
+export class ErrorHandlerService implements ErrorHandler {
+  dialogService: DialogService;
+  translateService: TranslateService;
+  get translate(): TranslateService {
```

Comment:
I'll address the rest of the comments. But the reason for this is that ErrorHandler is supposed to be ready before any other general services because all code naturally depends on ErrorHandler to be ready. However, adding dependencies in the ErrorHandler means now ErrorHandler is dependent on those services which results in a circular dependency. That's why, we get services instances lazily with the custom injection.

Reply:
Also, the service won't necessarily be available when the `ErrorHandlerService` service is constructed.